web/src/components/apps/AppDetailPage.tsx

```diff
@@ -99,8 +99,10 @@ function AppDetailPage(props: Props) {
     // navigate to first app if available
     if (appsList && appsList?.length > 0) {
       navigate(`/app/${appsList[0].slug}`, { replace: true });
-    } else if (Utilities.isLoggedIn()) {
+    } else if (Utilities.isLoggedIn() && !props.isEmbeddedCluster) {
       navigate("/upload-license", { replace: true });
+    } else if (Utilities.isLoggedIn && props.isEmbeddedCluster) {
```

Comment:
not sure if this affects the behavior you tested, but:
```suggestion
    } else if (Utilities.isLoggedIn() && props.isEmbeddedCluster) {
```

Reply:
@sgalsaleh fixed and tested.